.github/workflows/reposchutz.yml

```diff
@@ -0,0 +1,40 @@
+name: repository
+on:
+  pull_request_target:
+    types: [opened, reopened, synchronize, labeled, unlabeled]
+
+jobs:
+  check:
+    name: Protection checks
+    runs-on: ubuntu-latest
+    timeout-minutes: 1
+    env:
+      GIT_DIR: git-dir.git
+    steps:
+      - name: Clear .github-changes label
+        if: ${{ !endsWith(github.event.action, 'labeled') }}
+        uses: actions/github-script@v3
+        with:
+          script: |
+            try {
+              const { repo: { owner, repo }, number: issue_number } = context;
+              await github.issues.removeLabel({ owner, repo, issue_number, name: '.github-changes' });
+            } catch (e) {
+              if (e.name == 'HttpError' && e.status == 404) {
+                /* expected: 404 if label isn't set */
+              } else {
+                throw e;
+              }
+            }
+
+      - name: Check for .github changes
+        # We want to run this check any time the .github-changes label is not
+        # set, which needs to include the case where we just unset it above.
+        if: ${{ !endsWith(github.event.action, 'labeled') ||
+                !contains(github.event.pull_request.labels.*.name, '.github-changes') }}
+        run: |
+          set -x
+          git init -b main
+          git remote add origin "${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}"
+          git fetch origin '${{ github.event.pull_request.head.sha }}'
+          git diff --exit-code '${{ github.sha }}' '${{ github.event.pull_request.head.sha }}' -- .github
```

Comment:
So the intention here is to not fetch the whole repo (as a naïve `git clone` would do), but only the bits we need. I wonder how this can work, though, as it only fetches one half, and not `github.sha`. Either this only works if the PR is fully rebased on current master (and thus `github.sha` is part of the history of `github.event.pull_request.head.sha`), or somehow the fetch pulls all the heads/branches anyway (in which case there's nothing we can do).

Did you try this with an "outdated" branch?

Reply:
The `github.sha` is according to the docs, the base commit of the PR, which is also one of the parent commits of the merge commit that github makes for the PR (as I understand it).

But it also means I fetch the full history, which is not necessary.  Probably it makes sense to replace this with two separate fetches, of the SHAs that we are interested in, and `--depth=1`.

Reply:
Sounds good!